### DISCUSSION TITLE
Shelly Plus RGBW PM

### DISCUSSION THREAD

**Discussion by mewt666** [arendst/Tasmota#22703] [Support request] [answered]
I incorrectly assumed that the Shelly Plus RGBW PM was supported by Tasmota but I can't find any mention of the device anywhere. Is this supported in some way?

**Comment by arendst** [comment, marked as answer]
Supported. See minimal docs in source code file `xsns_02_analog.ino`

```
 * For shelly RGBW PM:
 * Template {"NAME":"Shelly Plus RGBW PM Pot.meter","GPIO":[1,0,0,0,419,0,0,0,0,0,544,0,0,0,0,1,0,0,1,0,0,416,417,418,0,0,0,0,0,4736,11264,11296,4704,0,4705,0],"FLAG":0,"BASE":1}
 * AdcGpio33 32000,40000,3350        <- Temperature parameters
 * AdcGpio34 1161,3472,10,30         <- Voltage parameters
 * AdcGpio35 960,1017,0.01,0.706     <- Current parameters
 * AdcGpio36 1552,176,3,1            <- I1 Potentiometer RGBW or RGB if SO37 128
 * AdcGpio36 1552,176,3,3            <- I1 Potentiometer RGBW or RGB if SO37 128 without fading
 * AdcGpio38 1552,176,3,1            <- I3 Potentiometer W if SO37 128
 * Fade 1
 * Speed 6
 * VoltRes 2
 * WattRes 2
 * 
 * Template {"NAME":"Shelly Plus RGBW PM Button","GPIO":[1,0,0,0,419,0,0,0,0,0,544,0,0,0,0,1,0,0,1,0,0,416,417,418,0,0,0,0,0,4736,11264,11296,4800,4801,4802,4803],"FLAG":0,"BASE":1}
 * AdcGpio33 32000,40000,3350        <- Temperature parameters
 * AdcGpio34 1161,3472,10,30         <- Voltage parameters
 * AdcGpio35 960,1017,0.01,0.706     <- Current parameters
 * AdcGpio36 511                     <- I1 ADC Button
 * AdcGpio37 511                     <- I2 ADC Button
 * AdcGpio38 511                     <- I3 ADC Button
 * AdcGpio39 511                     <- I4 ADC Button
 * Fade 1
 * Speed 6
 * VoltRes 2
 * WattRes 2
 * ```

**Reply by DDNRN** [reply]
Did you try it? 
.. and is it working?

**Reply by V8Mar** [reply]
Is it OK?

All switches are off and it shows about 4W.

<img width="350" height="439" alt="shelly plus rgbw" src="https://github.com/user-attachments/assets/7cbe9448-a97a-43e9-9026-872da6863111" />


It should probably be 0W, right?

And another problem:
My LED strip is 3.5 m long, and 15 W/1m. It's 52.5W, not 43.2W.
How to recalculate this?

<img width="361" height="376" alt="shelly plus rgbw _ALL ON" src="https://github.com/user-attachments/assets/6d5ba4f1-b809-4656-aad0-a1a77e8c86b0" />


AdcGpio34 1161,3472,10,30 <- Voltage parameters
AdcGpio35 960,1017,0.01,0.706 <- Current parameters

**Comment by mewt666** [comment]
Thank you greatly for your quick response. I will try to flash and hope for the best!

**Reply by sailmich** [reply]
I have a RGBW PM running with Tasmota and the suggested Template. Power monitoring doesn't work correct. I have five fans with PWM connected to the module. The power supply deliver 12V and 2A. The displayed Voltage is always 1,14V and Current 0,789A, doesen't matter if the fans are off or running with max speed. I assume that the Adc has to be part of the template. RGBW PM isn't on the supported device repository. When I compare the templates of other shelly PM devices, I can see a part in the templates about the Adc. E.c. {"NAME":"Shelly Plus 1PM","GPIO":[0,0,0,0,192,2720,0,0,0,0,0,0,0,0,2656,0,0,0,0,2624,0,32,224,0,0,0,0,0,4736,0,0,0,0,0,0,0],"FLAG":0,"BASE":1,"CMND":"AdcParam1 2,10000,10000,3350}
According to the tasmota docs I should compile my own build to activate Adc_VCC. But this is way above my skills :(.
Is someone already using RGBW PM with tasmota and get the right values?

**Reply by arendst** [reply]
You've read this whole thread? See above regarding templates and commands

**Reply by sailmich** [reply]
Thx for your rapid answer. Yes I read the whole thread. Seems I'm not smart enough to take the right conclusions. I believe the way you ask me, the answer is in your first reply. So I will try to find it.
